fix #57 .Env + Readme fixes (#58)
yetanotherco/yet-another-bridge: README.md

### Ethereum smart contract
First, the Ethereum smart contracts must be deployed. For Ethereum the deployment process you will need to:

1. Create your `.env` file: you need to configure the following variables in your own .env file on the contracts/solidity/ folder. You can use the env.example file as a template for creating your .env file, paying special attention to the formats provided

   ```bash
   ETH_RPC_URL = Infura or Alchemy RPC URL
   ETH_PRIVATE_KEY = private key of your ETH wallet
   ETHERSCAN_API_KEY = API Key to use etherscan to read the Ethereum blockchain
   SN_MESSAGING_ADDRESS = Starknet Messaging address
   ```
   **NOTE**:

   - You can generate ETHERSCAN_API_KEY [following this steps](https://docs.etherscan.io/getting-started/creating-an-account).
   - For the deploy, you will need some GoerliETH that you can get from this [faucet](https://goerlifaucet.com/).
   - Current SN_MESSAGING_ADDRESS values:
   - SN_MESSAGING_ADDRESS is for when a L1 contract initiates a message to a L2 contract on Starknet. It does so by calling the sendMessageToL2 function on the Starknet Core Contract with the message parameters. Starknet Core Contracts are the following:

>       - Mainnet 0xc662c410C0ECf747543f5bA90660f6ABeBD9C8c4
>       - Goerli: 0xde29d060D45901Fb19ED6C6e959EB22d8626708e
>       - Sepolia: 0xE2Bb56ee936fd6433DC0F6e7e3b8365C906AA057 

Reviewer: ```suggestion
      - Mainnet: `0xc662c410C0ECf747543f5bA90660f6ABeBD9C8c4`
      - Goerli: `0xde29d060D45901Fb19ED6C6e959EB22d8626708e`
      - Sepolia: `0xE2Bb56ee936fd6433DC0F6e7e3b8365C906AA057`
```

For this, you will need to:

1. Create your `.env` file: you need to configure the following variables in your own .env file on the contracts/solidity folder. You can use the env.example file as a template for creating your .env file, paying special attention to the formats provided

   ```bash
   STARKNET_ACCOUNT = Absolute path of your starknet testnet account, created at the start of this README
   STARKNET_KEYSTORE = Absolute path of your starknet testnet keystore, created at the start of this README
   SN_RPC_URL = Infura or Alchemy RPC URL
   ETH_CONTRACT_ADDR = newly created ETH contract address
   MM_SN_WALLET_ADDR = Starknet wallet of the MarketMaker
   WITHDRAW_NAME = The exact name of the function that withdraws from the starknet smart contract
   HERODOTUS_FACTS_REGISTRY = Herodotus' Facts Registry Smart Contract in Starknet
   MM_ETHEREUM_WALLET = Ethereum wallet of the MArketMaker
   NATIVE_TOKEN_ETH_STARKNET = Ethereum's erc20 token handler contract in Starkent
   ESCROW_CONTRACT_ADDRESS = Address of the Starknet smart contract, this value is automatically updated after deploy.sh is run
   ```

   **Note**
   - Herodotus Facts Registry:

>       - Starknet Goerli: 0x01b2111317EB693c3EE46633edd45A4876db14A3a53ACDBf4E5166976d8e869d
>       - Starknet Sepolia: 0x07d3550237ecf2d6ddef9b78e59b38647ee511467fe000ce276f245a006b40bc
>       - Starknet Mainnet: 0x014bf62fadb41d8f899bb5afeeb2da486fcfd8431852def56c5f10e45ae72765

Reviewer: ```suggestion
      - Starknet Goerli: `0x01b2111317EB693c3EE46633edd45A4876db14A3a53ACDBf4E5166976d8e869d`
      - Starknet Sepolia: `0x07d3550237ecf2d6ddef9b78e59b38647ee511467fe000ce276f245a006b40bc`
      - Starknet Mainnet: `0x014bf62fadb41d8f899bb5afeeb2da486fcfd8431852def56c5f10e45ae72765`
```

## Declare and Deploy Contracts in Testnet

### Ethereum smart contract
First, the Ethereum smart contracts must be deployed. For Ethereum the deployment process you will need to:

> 1. Create your `.env` file: you need to configure the following variables in your own .env file on the contracts/solidity/ folder. You can use the env.example file as a template for creating your .env file, paying special attention to the formats provided

Reviewer: > You can use the env.example file as a template for creating your .env file, paying special attention to the formats provided
Which format provided?

Author: The format provided is in the .env.example files

1. Create your `.env` file: you need to configure the following variables in your own .env file on the contracts/solidity folder. You can use the env.example file as a template for creating your .env file, paying special attention to the formats provided

   ```bash
   STARKNET_ACCOUNT = location of your starknet testnet account, created at the start of this README
   STARKNET_KEYSTORE = location of your starknet testnet keystore, created at the start of this README
   SN_RPC_URL = Infura or Alchemy RPC URL
   ETH_CONTRACT_ADDR = newly created ETH contract address
   MM_SN_WALLET_ADDR = Starknet wallet of the MarketMaker

   ```

2. Declare and Deploy: We sequentially declare and deploy the contracts.

   ```bash
      make starknet-deploy
   ```

### Finalize connection between both chains

After the Starknet smart contracts are declared and deployed, one variable from the Ethereum smart contract must be updated with the newly created Starknet smart contract address.

For now, to do this you must go to the ETH explorer and find your ETH smart contract, for example:
https://goerli.etherscan.io/address/0x8C78111CCDd368345c7d6A7bb6A7c887b1093f7A#writeContract

> Then, under the "contract/Write Contract" section, you must run the function, with your newly created Starknet escrow contract address as the parameter

Reviewer: We should add a way to do this by using foundry

Author: done, using ´cast send´

## Declare and Deploy Contracts in Testnet

### Ethereum smart contract
First, the Ethereum smart contracts must be deployed. For Ethereum the deployment process you will need to:

1. Create your `.env` file: you need to configure the following variables in your own .env file on the contracts/solidity/ folder. You can use the env.example file as a template for creating your .env file, paying special attention to the formats provided

   ```bash
   ETH_RPC_URL = Infura or Alchemy RPC URL
   ETH_PRIVATE_KEY = private key of your ETH wallet
   ETHERSCAN_API_KEY = API Key to use etherscan to read the Ethereum blockchain
   ```
   **NOTE**:

>    - You can generate ETHERSCAN_API_KEY [following this steps](https://docs.etherscan.io/getting-started/viewing-api-usage-statistics).

Reviewer: ```suggestion
   - You can generate ETHERSCAN_API_KEY [following this steps](https://docs.etherscan.io/getting-started/creating-an-account).
```

   ```bash
   ETH_RPC_URL = Infura or Alchemy RPC URL
   ETH_PRIVATE_KEY = private key of your ETH wallet
   ETHERSCAN_API_KEY = API Key to use etherscan to read the Ethereum blockchain
   ```
   **NOTE**:

   - You can generate ETHERSCAN_API_KEY [following this steps](https://docs.etherscan.io/getting-started/creating-an-account).
   - For the deploy, you will need some GoerliETH that you can get from this [faucet](https://goerlifaucet.com/).

2. Deploy Solidity contract

   ```bash
      make ethereum-deploy
   ```

### Starknet smart contracts

After the Ethereum smart contract is deployed, the Starknet smart contracts must be declared and deployed.
On Starknet, the deployment process is in two steps:

- Declaring the class of your contract, or sending your contract’s code to the
  network
- Deploying a contract or creating an instance of the previously declared code
  with the necessary parameters

For this, you will need to:

> 

Reviewer: We should add here the addresses for the Herodotus Fact Registry for Goerli, Sepolia and Mainnet.
You can take a look at this [website](https://docs.herodotus.dev/herodotus-docs/developers/contract-addresses) and we are already asking Herodotus for the Sepolia addresses.

Author: Did you also ask for the Starknet Mainnet addresses? They have only published Starknet Goerli Addresses

Reviewer: They just answered we can find them here:
 https://herodotus.notion.site/Contract-deployments-bfeacfc2d7b94c48aad688dbad553540

   ```bash
      make starknet-deploy
   ```

### Finalize connection between both chains

After the Starknet smart contracts are declared and deployed, one variable from the Ethereum smart contract must be updated with the newly created Starknet smart contract address.

For now, to do this you must go to the ETH explorer and find your ETH smart contract, for example:
https://goerli.etherscan.io/address/0x8C78111CCDd368345c7d6A7bb6A7c887b1093f7A#writeContract
Then, under the "contract/Write Contract" section, you must run the function, with your newly created Starknet escrow contract address as the parameter
```
snEscrowAddress(uint256)
```

This manual solution is temporary and wil be done automatically in the cairo deploy script.

To do this, you can also use Foundry's **cast** function, as follows:

```
cast send \
--rpc-url <eth_rpc_url> \
--private-key <your_private_key> \
<eth_smart_contract_address> "setEscrowAddress(uint256)" <your_new_SN_contract_address>
```

This will send a transaction, using the specified rpc, from the specified address, to the specified smart contract, calling the "setEscrowAddress" function, with your new SN contract address. 

> 

Reviewer 2: Here is missing the setEscrowWithdrawSelector process. If it is set automatically, add a comment about that

Author: Done
